lib/package/dynamodb/compileMethodsToDynamodb.js

```diff
@@ -0,0 +1,387 @@
+  },
+
+  buildDefaultDynamodbRequestTemplates(http) {
+    return {
+      'application/json': this.buildDefaultDynamodbRequestTemplate(http),
+      'application/x-www-form-urlencoded': this.buildDefaultDynamodbRequestTemplate(http)
+    }
+  },
+
+  getDynamodbObjectHashkeyParameter(http) {
+    if (http.hashKey.pathParam) {
+      return {
+        key: http.hashKey.pathParam,
+        attributeType: http.hashKey.attributeType,
+        attributeValue: `$input.params().path.${http.hashKey.pathParam}`
+      }
+    }
+
+    if (http.hashKey.queryStringParam) {
+      return {
+        key: http.hashKey.queryStringParam,
+        attributeType: http.hashKey.attributeType,
+        attributeValue: `$input.params().querystring.${http.hashKey.queryStringParam}`
+      }
+    }
+
+    if (http.hashKey.bodyParam) {
+      return {
+        key: http.hashKey.bodyParam,
+        attributeType: http.hashKey.attributeType,
+        attributeValue: `$util.parseJson($input.body).${http.hashKey.bodyParam}`
+      }
+    }
+
+    return {
+      key: http.hashKey,
+      attributeType: 'S',
+      attributeValue: '$context.requestId'
+    }
+  },
+
+  getDynamodbObjectRangekeyParameter(http) {
+    if (http.rangeKey.pathParam) {
+      return {
+        key: http.rangeKey.pathParam,
+        attributeType: http.rangeKey.attributeType,
+        attributeValue: `$input.params().path.${http.rangeKey.pathParam}`
+      }
+    }
+
+    if (http.rangeKey.queryStringParam) {
+      return {
+        key: http.rangeKey.queryStringParam,
+        attributeType: http.rangeKey.attributeType,
+        attributeValue: `$input.params().querystring.${http.rangeKey.queryStringParam}`
+      }
+    }
+
+    if (http.rangeKey.bodyParam) {
+      return {
+        key: http.rangeKey.bodyParam,
+        attributeType: http.rangeKey.attributeType,
+        attributeValue: `$util.parseJson($input.body).${http.rangeKey.bodyParam}`
+      }
+    }
+
+    return {
+      key: http.rangeKey,
+      attributeType: 'S',
+      attributeValue: '$context.requestId'
+    }
+  },
+
+  getDefaultDynamodbResponseTemplates(http) {
+    if (http.action === 'PutItem' && _.isString(http.hashKey)) {
+      return {
+        'application/json': this.getPutItemDefaultDynamodbResponseTemplate(http),
+        'application/x-www-form-urlencoded': this.getPutItemDefaultDynamodbResponseTemplate(http)
+      }
+    }
+
+    if (http.action === 'GetItem') {
+      return {
+        'application/json': this.getGetItemDefaultDynamodbResponseTemplate(),
+        'application/x-www-form-urlencoded': this.getGetItemDefaultDynamodbResponseTemplate()
+      }
+    }
+
+    return {}
+  },
+
+  getGetItemDefaultDynamodbResponseTemplate() {
+    return `#set($item = $input.path('$.Item')){#foreach($key in $item.keySet())#set ($value = $item.get($key))#foreach( $type in $value.keySet())"$key":"$value.get($type)"#if($foreach.hasNext()),#end#end#if($foreach.hasNext()),#end#end}`
```

Comment:
I think we can use https://github.com/declandewet/common-tags#onelinetrim to make this look nicer
import {oneLineTrim} from 'common-tags'

```
oneLineTrim`
  https://news.com/article
  ?utm_source=designernews.co
`
// https://news.com/article?utm_source=designernews.co
```